app/_kong_plugins/tls-metadata-headers/index.md

```diff
@@ -38,4 +38,41 @@ search_aliases:
   - certificates
 ---
 
-## Overview
+The TLS Metadata Header plugin detects client certificates in requests, extracts the TLS metadata (such as the URL-encoded client certificate), and injects this metadata into HTTP headers. It does not validate client certificates.
+
+Here are some use cases where the TLS Metadata Header plugin can be helpful:
+* Pass TLS client certificate metadata to an upstream service, enabling it to perform validation of the proxied certificate
+* Establish trust between microservices and use the extracted metadata to route requests differently based on the client’s certificate metadata (for example, different routes for different departments or services)
+* Enforce access control based on certain attributes of the client certificate, like the client’s organization (extracted from the certificate's subject DN)
+* Log or audit extracted metadata from client certificates
+
+{:.warning}
+> **Important:** This plugin **must** be used in conjunction with another plugin that requests a client certificate, such as the [mTLS Authentication](/plugins/mtls-auth/) or [TLS Handshake Modifier](/plugins/tls-handshake-modifier/) plugins. 
+
+## How it works
+
+The following explains how the TLS Metadata Headers plugin works:
+
+1. A client request is sent with a TLS certificate on a Gateway Service.
```

Comment:
```suggestion
1. A client sends a request with a TLS certificate.
```
"On a Gateway Service" seems out of place here. 
I'd also rephrase into less passive voice.